### PR TITLE
Add caveat concerning route code splitting in README.md

### LINKED ISSUE #1237: Named exports not working with async-loader
**Do you want to request a _feature_ or report a _bug_?**
This is a **Bug**

**What is the current behaviour?**
1/ Importing modules from `app.tsx` uses `async-loader` even if there is no `async!` prefix to it (why not)
2/ `async-loader` seems to not support named exports at all.

**If the current behaviour is a bug, please provide the steps to reproduce.**
```
src/
    - app.tsx
    routes/
        landing/
            - index.tsx
```

`app.tsx`
```
import { LandingPage } from 'routes/landing'
class App extends Components {

...

render() {
    return (
        ...
        <Router>
            ...
            <Route component={LandingPage}>
            ...
        </Router>
        ...
    )
}

export default App
```

`landing/index.tsx`
```
import { FunctionalComponent, h } from 'preact'
import LandingPageLegacy from 'legacy/src/core/js/forum/components/LandingPage'
import { LegacyComponent } from 'legacy/compat'

export const LandingPage: FunctionalComponent = () => {
  return (
    <div>
      <LegacyComponent component={LandingPageLegacy} />
    </div>
  )
}

export default LandingPage
```
-----
**1. Named export does not work**
```
⚠ WARN ./app.tsx 140:17-28
public_frontend_1  | "export 'LandingPage' was not found in 'routes/landing'
public_frontend_1  |  @ ./index.js
public_frontend_1  |  @ ../node_modules/preact-cli/lib/lib/entry.js
public_frontend_1  |  @ multi ../node_modules/preact-cli/lib/lib/entry webpack-dev-server/client webpack/hot/dev-server
```
A `console.log(LandingPage)` at runtime yields `undefined`

**2. Switching back to default import `import LandingPage from 'routes/landing'` does work**
**3. Adding this**
```
import * as test from 'routes/landing'

...

class App extends Component ‘{
    ...
    render() {
        ...
        console.log(test)
        ...
    }
}
```
Logs:
image

At other places in my codebase this works fine because the `async-loader` is not doing the job.

**What is the expected behaviour?**
One of these
- Have the named exports working (prefered)
- Have it documented somewhere (didn't find any mention of this behavior)

### PR DESCRIPTION
**Pull request by toniopelo**
Fixes #1237

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -30,7 +30,7 @@
 ### Features
 
 - **100/100 Lighthouse score**, right out of the box ([proof])
-- Fully **automatic code splitting** for routes
+- Fully **automatic code splitting** for routes (see caveat in [Route automatic code splitting](#route-automatic-code-splitting) section)
```
Comment (developit) [inline README.md:33]:
```suggestion
- Fully **automatic code splitting** for routes _(see [Route-Based Code Splitting](#route-based-code-splitting))_
```

File: README.md [hunk 2]
```diff
@@ -366,3 +366,17 @@ The default templates comes with a `.css` file for each component. You can start
 [`async!`]: https://github.com/preactjs/preact-cli/blob/1.4.1/examples/full/src/components/app.js#L7
 [sass]: https://sass-lang.com
 [less]: http://lesscss.org
+
+### Route automatic code splitting
```
Comment (developit) [inline README.md:370]:
```suggestion
### Route-Based Code Splitting
```

File: README.md [hunk 3]
```diff
@@ -366,3 +366,17 @@ The default templates comes with a `.css` file for each component. You can start
...
+
+Route code splitting works with the [async-loader](https://github.com/preactjs/preact-cli/tree/master/packages/async-loader).
+This feature will be automatically enabled for files located in:
+- `src/routes/{*,*/index}.{js,jsx,ts,tsx}`
+- `src/components/routes/{*,*/index}.{js,jsx,ts,tsx}`
+- `src/components/async/{*,*/index}.{js,jsx,ts,tsx}`
```
Comment (developit) [inline README.md:376]:
```suggestion
"Route" components are automatically code-splitted at build time to create smaller bundles and avoid loading more code than is needed by each page. This works by intercepting imports for route components with an [async loader](https://github.com/preactjs/preact-cli/tree/master/packages/async-loader), which returns a lightweight wrapper component that handles code splitting seamlessly.

Automatic code splitting is applied to all JavaScript and TypeScript files in the following locations:


<table>
<thead><tr><th>Pattern</th><th>Examples</th></tr></thead>
<tbody>
<tr><td>
<pre>src/<b>routes</b>/<kbd>NAME</kbd></pre>
</td><td>
<code>src/routes/home.js</code><br>
<code>src/routes/about/index.tsx</code>
</td></tr>
<tr><td>
<pre>src/components/<b>routes</b>/<kbd>NAME</kbd></pre>
</td><td>
<code>src/components/routes/profile.ts</code><br>
<code>src/components/routes/profile/index.js</code>
</td></tr>
<tr><td>
<pre>src/components/<b>async</b>/<kbd>NAME</kbd></pre>
</td><td>
<code>src/components/async/profile.ts</code><br>
<code>src/components/async/profile/index.js</code>
</td></tr>
</tbody></table>
```

File: README.md [hunk 4]
```diff
@@ -366,3 +366,17 @@ The default templates comes with a `.css` file for each component. You can start
...
+
+**Caveat**:
+
+The async loader can only work with default exports.
+Make sure to have your component exported as default or you module import will be broken at runtime (Imported component will be undefined).\
+If you wish to disable this behavior, rename your folder to avoid matching the above paths.
```
Comment (developit) [inline README.md:382]:
```suggestion
> **Note**:  
> Automatic code splitting **only** supports default exports, not named exports:
>
> ```diff
> - import { Profile } from './routes/profile';
> + import Profile from './routes/profile';
> ```
>
> This is an intentional limitation that ensures effective code splitting. For components that need named exports, place them in a directory that doesn't trigger automatic code splitting. You can then manually code-split the default export by re-exporting it from `routes/` or importing it with the `"async!"` prefix.
```

**Comment by developit** [comment]
Echoing from chat: there was actually a super valid concern in the original text that I think we can just fix outright: that things might work during dev but fail in prod. We can easily change the development mode to have the same behavior as production for these directories: instead of replacing `async-loader` with a passthrough during development, we can replace it with a passthrough that _only re-exports the default export_.

**Comment by toniopelo** [comment]
Just committed your suggestions @developit :)

**Pull request merged**